prog/weaveproxy/main.go

```diff
@@ -85,3 +87,44 @@ func main() {
 
 	p.ListenAndServe()
 }
+
+func loadDefaultListenAddr(c *proxy.Config) {
+	if len(c.ListenAddrs) > 0 {
+		return
+	}
+
+	clientTLSEnabled := (os.Getenv("DOCKER_CLIENT_TLS_VERIFY") != "")
+
+	clientArgs := strings.Fields(os.Getenv("DOCKER_CLIENT_ARGS"))
+	filteredArgs := []string{}
+	for i := 0; i < len(clientArgs); i++ {
+		arg := strings.SplitN(clientArgs[i], "=", 2)
+		switch arg[0] {
+		case "-tls", "--tls", "-tlsverify", "--tlsverify":
+			clientTLSEnabled = true
+		case "-H", "--host":
+			filteredArgs = append(filteredArgs, clientArgs[i])
+			if !strings.Contains(clientArgs[i], "=") {
+				if i++; i < len(clientArgs) {
+					filteredArgs = append(filteredArgs, clientArgs[i])
+				}
+			}
+		}
+	}
+
+	if clientTLSEnabled && !(c.TLSConfig.Enabled || c.TLSConfig.Verify) {
+		log.Fatalln("Cannot autoconfigure proxy listen address when launching via TLS")
```

Comment:
That error message is still misleading. We *do* autoconfigure the proxy listen address. What we *don't* do is configure TLS.